src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -177,26 +177,23 @@ public void removeAclOnFiles(Dataset dataset, List<String> fileIds, String reade
         fileAclOp(AclOp.ACL_OP_DELETE, dataset, fileIds, readersPolicyEmail);
     }
 
-    public static Blob getBlobFromGsPath(Storage storage, String gspath) {
-        String sourceBucket;
-        String sourcePath;
+    private static final String gsProtocol = "gs://";
 
-        try {
-            URI sourceUri = URI.create(gspath);
-            if (!StringUtils.equals(sourceUri.getScheme(), "gs")) {
-                throw new PdaoInvalidUriException("Path is not a gs path: '" + gspath + "'");
-            }
-            if (sourceUri.getPort() != -1) {
-                throw new PdaoInvalidUriException("Path must not have a port specified: '" + gspath + "'");
-            }
-            sourceBucket = sourceUri.getAuthority();
-            sourcePath = StringUtils.removeStart(sourceUri.getPath(), "/");
-        } catch (IllegalArgumentException ex) {
-            throw new PdaoInvalidUriException("Invalid gs path: '" + gspath + "'", ex);
+    public static Blob getBlobFromGsPath(Storage storage, String gspath) {
+        if (!StringUtils.startsWith(gspath, gsProtocol)) {
+            throw new PdaoInvalidUriException("Path is not a gs path: '" + gspath + "'");
         }
 
-        if (sourceBucket == null || sourcePath == null) {
-            throw new PdaoInvalidUriException("Invalid gs path: '" + gspath + "'");
+        String noGsUri = StringUtils.substring(gspath, gsProtocol.length());
+        String sourceBucket = StringUtils.substringBefore(noGsUri, "/");
```

Comment:
The old code had some bucket name validation. Should we have some check that makes sure the bucket name at least obeys the GCS naming rules? Or catch an error from the getBlob?

Reply:
:+1: will add more checks

Reply:
Covered everything I could see from the GCS docs